src/config.rs

```diff
@@ -12,15 +12,17 @@ use crate::utils::serde_utils::JsonStrWrapper;
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Config {
-    pub app:       AppConfig,
-    pub tree:      TreeConfig,
-    pub network:   NetworkConfig,
-    pub providers: ProvidersConfig,
-    pub relayer:   RelayerConfig,
-    pub database:  DatabaseConfig,
-    pub server:    ServerConfig,
+    pub app:           AppConfig,
+    pub tree:          TreeConfig,
+    pub network:       NetworkConfig,
+    pub providers:     ProvidersConfig,
+    pub relayer:       RelayerConfig,
+    pub database:      DatabaseConfig,
+    pub server:        ServerConfig,
     #[serde(default)]
-    pub service:   ServiceConfig,
+    pub service:       ServiceConfig,
+    #[serde(default)]
+    pub offchain_mode: OffchainModeConfig,
```

Comment:
I feel like with off-chain mode we should make network, providers and relayer configs optional

Reply:
Done